src/main/java/org/opensearch/commons/destination/message/LegacyChimeMessage.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.commons.destination.message;
+
+import java.io.IOException;
+
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+
+/**
+ * This class holds the contents of an Chime message
+ */
+public class LegacyChimeMessage extends LegacyBaseMessage {
```

Comment:
is there any reason why we cannot use Chime.kt and ChannelMessage.kt classes and create a legacy class similar to SendMessageRequest.kt ? wouldn't this reduce all these duplication of code? (similarly for other destination types as well)

Reply:
These classes are copied over from the alerting notification subproject. The reason why am I trying to keep them as close to same is:
1. Already been tested in production for 2+ years.
2. By keeping them separate it allows the notification plugin to continue to extend and make changes to your own channel classes without worrying about ISM legacy destinations. I don't want your team to worry about making changes to Slack.kt or Chime.kt and have to think about the legacy ISM use case. These are essentially just copy pasted over and should not be touched ever at this point.